prog/dftb+/lib_dftb/rangeseparated.F90

```diff
@@ -293,7 +293,11 @@ subroutine updateCoordsNeigh(this, coords, nNeigh, iNeigh, img2CentCell)
 
     nAt = size(this%species)
     @:ASSERT(nAt == size(nNeigh))
-    this%coords(:,:) = coords(:, :nAt)
+    if (size(this%coords, dim=2) /= size(coords, dim=2)) then
+      deallocate(this%coords)
+      allocate(this%coords(3,size(coords, dim=2)))
+    end if
+    this%coords(:,:) = coords
```

Comment:
Why not just use the automatic reallocation of the LHS directly rather than implementing it here?